rosbag2_tests/test/rosbag2_tests/test_rosbag2_play_end_to_end.cpp

```diff
@@ -55,6 +56,17 @@ class PlayEndToEndTestFixture : public Test, public WithParamInterface<std::stri
       [this]() {
         exec_->spin();
       });
+
+    // Wait for the executor to start spinning in the newly spawned thread to avoid race conditions
+    using clock = std::chrono::steady_clock;
+    auto start = clock::now();
+    while (!exec_->is_spinning() && (clock::now() - start) < std::chrono::seconds(5)) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(20));
+    }
+    if (!exec_->is_spinning()) {
+      std::cerr << "Failed to start spinning node" << std::endl;
+      throw std::runtime_error("Failed to start spinning node");
+    }
```

Comment:
This looks like it would make a good test function (e.g., in a base test fixture class) that could be used in multiple places.

I'll let you decide whether or not to do that.

Reply:
Thanks for catching this.  
I replaced some boilerplates with 
```cpp
if (!wait_until_condition([this]() {return exec_->is_spinning();}, std::chrono::seconds(5))) {
```